File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task17.java

public class Task17 {
    public static String toJadenCase(String phrase) {
        if (phrase == null || phrase.isEmpty()) {
            return null;
        }
        
        StringBuilder result = new StringBuilder();
        boolean capitalizeNext = true;
        
        for (int i = 0; i < phrase.length(); i++) {
            char c = phrase.charAt(i);
            
            if (Character.isWhitespace(c)) {
                result.append(c);
                capitalizeNext = true;
            } else if (capitalizeNext && Character.isLetter(c)) {
                result.append(Character.toUpperCase(c));
                capitalizeNext = false;
            } else {
                result.append(c);
            }
        }
        
        return result.toString();
    }
    
    public static void main(String[] args) {
        String[] testCases = {
            "How can mirrors be real if our eyes aren't real",
            "hello world",
            "i love java programming",
            "it's a beautiful day",
            null
        };
        
        for (int i = 0; i < testCases.length; i++) {
            String result = toJadenCase(testCases[i]);
            System.out.println("Test " + (i + 1) + ": " + result);
        }
    }
}

File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task17.py

def to_jaden_case(phrase: str) -> str:
    if phrase is None or phrase == "":
        return None
    
    result = []
    capitalize_next = True
    
    for char in phrase:
        if char.isspace():
            result.append(char)
            capitalize_next = True
        elif capitalize_next and char.isalpha():
            result.append(char.upper())
            capitalize_next = False
        else:
            result.append(char)
    
    return ''.join(result)


def main():
    test_cases = [
        "How can mirrors be real if our eyes aren't real",
        "hello world",
        "i love python programming",
        "it's a beautiful day",
        None
    ]
    
    for i, test in enumerate(test_cases, 1):
        result = to_jaden_case(test)
        print(f"Test {i}: {result}")


if __name__ == "__main__":
    main()
